fee_allocator/accounting/distribution.py

```diff
@@ -119,21 +118,43 @@ def re_distribute_incentives(
             incentives[pool_id]['aura_incentives'] = 0
             incentives[pool_id]['bal_incentives'] += incentives_to_redistribute
             debt_to_aura_market += incentives_to_redistribute
-
+    # We have now allocated all the funds to the pools that deserve it.
+    # We have also potentially moved some value to the Balancer market on pools under the min_aura_incentive
+    # As a result we have generated some debt that we owe the aura market in order to maintain the AURA/BAL split
+    # So now we find pools that have room to reallocate and adjust their split to restore the balance
     if debt_to_aura_market:
-        amount_per_pool =  round(debt_to_aura_market / 6, 4)
-        # Create an array of the top 6 pools that received the most total_incentives
-        top_6_pools = sorted(incentives.items(), key=lambda x: x[1]['total_incentives'], reverse=True)[:6]
-        # Distribute  1/6th of incentives_taken_from_aura_market to each of the top 6 pools, while subtracting the same amount from bal_incentives
-        for pool_id, _data in top_6_pools:
-            if _data['total_incentives'] > 0:
-                # TODO:  Still an edge case here.  If amount per pool is not allocated, we'll still underallocate to
-                #   aura.  It shouldn't happen on the 6 largest pools system wide though.
-                incentives[pool_id]['aura_incentives'] += min(amount_per_pool, _data['bal_incentives'])
-                incentives[pool_id]['bal_incentives'] -= min(amount_per_pool, _data['bal_incentives'])
+        debt_repaid = 0
+        ## Find how many pools ever could be over min_aura_incentive
+        pools_over_aura_min = [pool_id for pool_id, _data in incentives.items() if
+                               _data['aura_incentives'] >= min_aura_incentive]
+        num_pools_over_min = len(pools_over_aura_min)
+        ## Figure out how much to shift per pool using an even split
+        if num_pools_over_min == 0:
+            print(
+                f"WARNING: {incentives[pool_id]['chain']}:{pool_id} has no pools over min_aura_incentive, but owes {debt_to_aura_market} to the aura market.  Debt will not be repaid.")
```

Comment:
is it worth it to `raise` here instead? how big of an erroneous csv can this produce?

Reply:
The final check in the recon (that we adjusted from 20 to 5) will make sure that we don't get too far away from where we should be.  If we do we have to refactor more.  Hoping to refactor this whole thing before that happens.

Reply:
got it